### PR TITLE
display explanations of modifications made to URDF

### PR DESCRIPTION
**Pull request by lewandowskikacper**
Proposition of a way to present modifications of URDF to user:
Resolves https://github.com/o3de/o3de-extras/issues/512.

Note: this PR includes commits from https://github.com/o3de/o3de-extras/pull/511, and should be merged after it.

image

### CODE REVIEW

**Comment by michalpelka** [comment]
Please rebase and then set it to review. It is an important change. @lewandowskikacper

File: Gems/ROS2/Code/Source/RobotImporter/URDF/UrdfParser.h [hunk 1]
```diff
@@ -75,8 +76,8 @@ namespace ROS2
             //! Stores the modified URDF content after parsing, empty if no modification occurred
             std::string m_modifiedURDFContent;
 
-            //! Stores the modified URDF tags after parsing, empty if no modification occurred
-            AZStd::vector<AZStd::string> m_modifiedURDFTags;
+            //! Stores describtion of URDF modifications, empty if no modification occurred
+            Utils::UrdfModifications m_UrdfModifications;
```
Comment (jhanca-robotecai) [inline Gems/ROS2/Code/Source/RobotImporter/URDF/UrdfParser.h:80]: The same change applies for the rest of usages
```suggestion
            Utils::UrdfModifications m_urdfModifications;
```

**Review by jhanca-robotecai — COMMENTED** [review]
The changes are looking good, please continue.

File: Gems/ROS2/Code/Source/RobotImporter/FixURDF/FixURDF.cpp [hunk 2]
```diff
@@ -41,23 +43,36 @@ namespace ROS2::Utils
...
+            incompleteInertia.linkName = name;
+
             auto mass = inertial->first_node("mass");
             if (!mass)
             {
-                AZ_Warning("URDF", !inertiaPresent, "Missing mass tag inside the inertial tag link %s, applying default mass tag.", name.c_str());
+                AZ_Warning(
+                    "URDF", !inertialPresent, "Missing mass tag inside the inertial tag link %s, applying default mass tag.", name.c_str());
                 mass = urdf->document()->allocate_node(node_element, "mass");
                 mass->append_attribute(urdf->document()->allocate_attribute("value", "1."));
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/RobotImporter/FixURDF/FixURDF.cpp:62]: There could be an estimation made from the mesh interior volume here (general comment, I am aware PR does not change that).

File: Gems/ROS2/Code/Source/RobotImporter/RobotImporterWidget.cpp [hunk 3]
```diff
@@ -159,10 +160,36 @@ namespace ROS2
                 {
                     // This is a URDF only path, and therefore the report text does not mention SDF
```
[the remark this answers is not in the document]
Comment (lewandowskikacper) [reply Gems/ROS2/Code/Source/RobotImporter/RobotImporterWidget.cpp]: Moved to `AddModificationWarningsToReportString`

**Review by michalpelka — APPROVED** [review]

**Review by adamdbrw — APPROVED** [review]

**Comment by lewandowskikacper** [comment]
I rebased on top of development again. Please note that without https://github.com/o3de/o3de-extras/pull/534 it will not compile.

I also noticed that "The following joints were renamed to avoid duplication" is printed even if no joint was renamed (but e.g. some inertial was missing). I added a simple check for that.

**Pull request merged**